File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/amap_to_eal.cpp
#include <iostream>
#include <vector>
using namespace std;

// Dinh nghia cau truc canh
struct Edge {
    int u, v, w;
};

int main() {
    int n, num_keys;
    cout << "Nhap so dinh: ";
    cin >> n;
    cout << "Nhap so dinh co trong AMap: ";
    cin >> num_keys;

    vector<Edge> eal; // Danh sach canh

    cout << "Nhap AMap (u, so dinh ke, cac cap v w):\n";
    for (int i = 0; i < num_keys; ++i) {
        int u, k;
        cin >> u >> k;
        for (int j = 0; j < k; ++j) {
            int v, w;
            cin >> v >> w;
            eal.push_back({u, v, w}); // Them canh vao danh sach
        }
    }

    cout << "Danh sach canh (EAL):\n";
    for (const Edge& e : eal) {
        cout << "(" << e.u << ", " << e.v << ", " << e.w << ")\n";
    }

    return 0;
}

File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/amap_to_eal.py
n = int(input("Nhap so dinh: "))
num_keys = int(input("Nhap so dinh co trong AMap: "))

eal = []  # Danh sach canh (Edge Array List)

print("Nhap AMap (u, so dinh ke, cac cap v w):")
for _ in range(num_keys):
    parts = input().split()
    u = int(parts[0])
    k = int(parts[1])
    for i in range(k):
        v = int(parts[2 + 2 * i])
        w = int(parts[3 + 2 * i])
        eal.append((u, v, w))  # Them canh (u, v, w) vao danh sach

# In danh sach canh
print("Danh sach canh (EAL):")
for edge in eal:
    print(edge)